From a0e6d634b37391b3ecf5d74f7e2092063df61198 Mon Sep 17 00:00:00 2001
From: mayankkk <mayankkk@google.com>
Date: Tue, 3 Dec 2024 17:36:14 -0800
Subject: [PATCH 02/10] Don't allow non-system uids to use "android" as calling
 package.

Bug: 310632322
Test: atest ./BroadcastsSecurityTest/src/android/app/cts/broadcasts/BroadcastRegistrationTest.java
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:84e975cf12759d04863e35abf51a94356c6b4c25)
Merged-In: I83537fde6cad2073e767d56c27cd25b82f29c235
Change-Id: I83537fde6cad2073e767d56c27cd25b82f29c235
---
 .../java/com/android/server/am/ActivityManagerService.java     | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 9a9c05060a4c2..5d2c13822cd28 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -12484,8 +12484,7 @@ public class ActivityManagerService extends IActivityManager.Stub
                             + ") when registering receiver " + receiver);
                 }
                 if (callerApp.info.uid != SYSTEM_UID
-                        && !callerApp.getPkgList().containsKey(callerPackage)
-                        && !"android".equals(callerPackage)) {
+                        && !callerApp.getPkgList().containsKey(callerPackage)) {
                     throw new SecurityException("Given caller package " + callerPackage
                             + " is not running in process " + callerApp);
                 }
-- 
2.40.1

